argilla/src/argilla/suggestions.py

```diff
@@ -119,15 +120,20 @@ def api_model(self) -> SuggestionModel:
             return self._model
 
         question = self.record.dataset.settings.questions[self.question_name]
-        return SuggestionModel(
-            value=self.__to_model_value(self.value, question),
-            question_name=question.name,
-            question_id=question.id,
-            type=self._model.type,
-            score=self._model.score,
-            agent=self._model.agent,
-            id=self._model.id,
-        )
+        if question:
+            return SuggestionModel(
+                value=self.__to_model_value(self.value, question),
+                question_name=None if not question else question.name,
+                question_id=None if not question else question.id,
+                type=self._model.type,
+                score=self._model.score,
+                agent=self._model.agent,
+                id=self._model.id,
+            )
+        else:
+            raise SuggestionsError(
+                f"suggestion invalid because targeted question with name={self.question_name} does not exist. Available questions are: {list(self.record.dataset.settings.questions._properties_by_name.keys())}"
```

Comment:
```suggestion
                f"Record suggestion is invalid because question with name={self.question_name} does not exist in the dataset ({self._record.dataset.name}). Available questions are: {list(self.record.dataset.settings.questions._properties_by_name.keys())}"
```